public_tests

LikeARivers opened the pull request:
публикую тесты

LikeARivers committed:
public_tests

AlexKitIt commented on pages/main_page.py:
```diff
@@ -0,0 +1,143 @@
+    QUEST_8 = (By.XPATH, "//div[@id = 'accordion__heading-7']")
+
+    @allure.step('скролл до Вопроса №1')
+    def scroll_to_quest_1(self, driver):
```
Нужно исправить: здесь и далее: driver не передается в методы, он является атрибутом пейджи

AlexKitIt commented on pages/main_page.py:
```diff
@@ -0,0 +1,143 @@
+    @allure.step('скролл до Вопроса №1')
+    def scroll_to_quest_1(self, driver):
+        quest_1 = (self.wait_and_find_element(self.QUEST_1))
+        driver.execute_script("arguments[0].scrollIntoView();", quest_1)
```
Нужно исправить: здесь и далее по проекту: все прямые обращения к driver и waitы необходимо вынести в base и оттуда за счет наследования переиспользовать в объектах страниц

AlexKitIt commented on pages/order_pages_1.py:
```diff
@@ -0,0 +1,90 @@
+    @allure.step('в поле Телефон вводим +7123456789')
+    def input_phone_number(self):
+        phone_number = (self.wait_and_find_element(self.INPUT_PHONE_NUMBER))
+        phone_number.send_keys('+7123456789')
```
Можно лучше: тестовые данные стоит хранить отдельно в data модуле, так поддерживать и переиспользовать будет проще

AlexKitIt commented on pages/order_pages_2.py:
```diff
@@ -0,0 +1,106 @@
+from selenium.webdriver.common.by import By
+from pages.base_page import BasePage
+
+class PagesPlacingAnOrder(BasePage):
```
Нужно исправить: класс с таким именем уже есть, нейминги должны различаться

AlexKitIt commented on tests/test_bottom_button_path.py:
```diff
@@ -0,0 +1,19 @@
+
+        order_pages.scroll_to_bottom_button(driver)
+        order_pages.click_bottom_button()
+        current_url = driver.current_url
```
Нужно исправить:  здесь и далее: прямого обращения к driver из тестов быть не должно , только через методы пейджей

AlexKitIt commented on tests/test_quest_1.py:
```diff
@@ -0,0 +1,24 @@
+from pages.main_page import MainPageQuests, MainPageAnswers
+
+
+class TestQuest1:
```
Нужно исправить: тест на вопросы должен быть один параметризованный(на вход примет элемент вопроса, элемент ответа и текст ответа)

AlexKitIt commented on tests/test_order_2.py:
```diff
@@ -0,0 +1,33 @@
+
+
+
+class TestOrder2:
```
Можно лучше: не надо создавать отдельные модули под каждый тест, тесты заказа - один модуль, один класс, тесты редиректа - так же, вопросы - так же

LikeARivers committed:
public_tests
убраны прямые обращения к driver из тестов, только через page
драйвер больше не передается в методы, является атрибутом page

LikeARivers committed:
public_tests

убраны прямые обращения к driver из тестов, только через page
драйвер больше не передается в методы, является атрибутом page

все прямые обращения к driver вынесены в base и оттуда за счет наследования переиспользованы в объектах страниц

LikeARivers committed:
public_tests

убраны прямые обращения к driver из тестов, только через page
драйвер больше не передается в методы, является атрибутом page

все прямые обращения к driver вынесены в base и оттуда за счет наследования переиспользованы в объектах страниц

в order_pages_1.py и order_pages_2.py изменен неймиинг классов

LikeARivers committed:
public_tests

убраны прямые обращения к driver из тестов, только через page
драйвер больше не передается в методы, является атрибутом page

все прямые обращения к driver вынесены в base и оттуда за счет наследования переиспользованы в объектах страниц

в order_pages_1.py и order_pages_2.py изменен неймиинг классов

применена параметризация в файле test_quests_and_answers

LikeARivers committed:
public_tests

убраны прямые обращения к driver из тестов, только через page
драйвер больше не передается в методы, является атрибутом page

все прямые обращения к driver вынесены в base и оттуда за счет наследования переиспользованы в объектах страниц

в order_pages_1.py и order_pages_2.py изменен неймиинг классов

применена параметризация в файле test_quests_and_answers

LikeARivers committed:
public_tests

убраны прямые обращения к driver из тестов, только через page
драйвер больше не передается в методы, является атрибутом page

все прямые обращения к driver вынесены в base и оттуда за счет наследования переиспользованы в объектах страниц

в order_pages_1.py и order_pages_2.py изменен неймиинг классов

применена параметризация в файле test_quests_and_answers

AlexKitIt commented on pages/main_page.py:
```diff
@@ -0,0 +1,38 @@
+        quest = self.wait_and_find_element(quest_element)
+        quest.click()
+
+class MainPageAnswers(BasePage):
```
Это одна и та же пейджа, не надо так из разбивать, это не pageElement

AlexKitIt commented on pages/order_pages_2.py:
```diff
@@ -0,0 +1,106 @@
+from selenium.webdriver.common.by import By
+from pages.base_page import BasePage
+
+class PagesPlacingAnOrder2(BasePage):
```
Нужно исправить: класс - полная копия предыдущего. Пейджи никак не завязаны на тестах, в которых используется. Объект страницы - это одна конкретная страница, тот факт, что мы передаем разные данные для заполнения заказа не говорит о том, что надо копипастить пейджу, а то, что надо делать в методах принимаемые значения. Пейджу нужно снести, добавить возможность принимать значения в методы предыдущей педйжи

AlexKitIt commented on tests/test_order_2.py:
```diff
@@ -0,0 +1,33 @@
+import allure
+from data import URL
+from pages.order_pages_2 import PagesPlacingAnOrder2
+import time
```
Нужно исправить: неиспользуемый импорт

LikeARivers committed:
удален неиспользуемый import time из test_order_2.py

LikeARivers committed:
удален неиспользуемый import time из test_order_2.py

элементы QUEST и ANSWER перенесены в один класс

LikeARivers committed:
удален неиспользуемый import time из test_order_2.py

элементы QUEST и ANSWER перенесены в один класс

файлы order_pages_1.py и order_pages_2.py объединены в один файл order_pages.py

LikeARivers committed:
удален неиспользуемый import time из test_order_2.py

элементы QUEST и ANSWER перенесены в один класс

файлы order_pages_1.py и order_pages_2.py объединены в один файл order_pages.py

проведены allure тесты

AlexKitIt commented on pages/order_pages.py:
```diff
@@ -0,0 +1,148 @@
+        list_subway = self.wait_and_find_element(self.LIST_SUBWAY)
+        list_subway.click()
+
+    @allure.step('в поле Телефон вводим +7123456789')
+    def input_phone_short_number(self):
+        phone_number = self.wait_and_find_element(self.INPUT_PHONE_NUMBER)
+        phone_number.send_keys('+7123456789')
+
+    @allure.step('в поле Телефон вводим +712312312312')
+    def input_phone_long_number(self):
+        phone_number = self.wait_and_find_element(self.INPUT_PHONE_NUMBER)
+        phone_number.send_keys('+712312312312')
```
Нужно  исправить: не так, я же указал - принимаемым значением, в тесте будем данные туда скармливать а не в пейдже. Метод помимо self примет номер телефона
Все дубли методов необходимо удалить и передавать значения для заполнения полей из параметров теста или писать два отдельных теста, но передавать в тесте
def input_phone_long_number(self, number):
        phone_number = self.wait_and_find_element(self.INPUT_PHONE_NUMBER)
        phone_number.send_keys(number)

LikeARivers committed:
исправлены файлы order_pages.py, test_order_1.py, test_order_2.py
